go/libraries/doltcore/sqle/enginetest/dolt_queries_schema_merge.go

```diff
@@ -2003,3 +2094,236 @@ var SchemaChangeTestsSchemaConflicts = []MergeScriptTest{
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 1)",
+			"insert into t (pk) values (3);",
+			"alter table t add index idx1 (col1, pk);",
+			"alter table t add index idx2 (col1);",
+		},
+		LeftSetUpScript: []string{
+			"alter table t add column col2 int;",
+			"alter table t add column col3 int;",
+			"insert into t (pk, col2, col3) values (2, 4, 5);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+			},
+			{
+				Query: "select pk, col1, col2, col3 from t;",
+				Expected: []sql.Row{
+					{1, 2, nil, nil},
+					{2, 3, 4, 5},
+					{3, 4, nil, nil},
+				},
+			},
+		},
+	},
+	{
+		Name: "adding a virtual column to one side",
+		AncSetUpScript: []string{
+			"create table t (pk int primary key);",
+			"insert into t (pk) values (1);",
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 1)",
+			"insert into t (pk) values (3);",
+			"alter table t add index idx1 (col1, pk);",
+			"alter table t add index idx2 (col1);",
+		},
+		LeftSetUpScript: []string{
+			"insert into t (pk) values (2);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+			},
+			{
+				Query: "select pk, col1 from t;",
+				Expected: []sql.Row{
+					{1, 2},
+					{2, 3},
+					{3, 4},
+				},
+			},
+		},
+	},
+	{
+		Name: "adding a stored generated column to one side",
+		AncSetUpScript: []string{
+			"create table t (pk int primary key);",
+			"insert into t (pk) values (1);",
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 1) stored",
+			"insert into t (pk) values (3);",
+			"alter table t add index idx1 (col1, pk);",
+			"alter table t add index idx2 (col1);",
+		},
+		LeftSetUpScript: []string{
+			"insert into t (pk) values (2);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+			},
+			{
+				Query: "select pk, col1 from t;",
+				Expected: []sql.Row{
+					{1, 2},
+					{2, 3},
+					{3, 4},
+				},
+			},
+		},
+	},
+	{
+		Name: "adding generated columns to both sides",
+		AncSetUpScript: []string{
+			"create table t (pk int primary key);",
+			"insert into t values (1), (2);",
+		},
+		RightSetUpScript: []string{
+			"alter table t add column col2 varchar(100) as (concat(pk, 'hello'));",
+			"insert into t (pk) values (3), (4);",
+			"alter table t add index (col2);",
+		},
+		LeftSetUpScript: []string{
+			"alter table t add column col1 int as (pk + 100) stored;",
+			"insert into t (pk) values (5), (6);",
+			"alter table t add index (col1);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "call dolt_merge('right');",
+				Expected: []sql.Row{{doltCommit, 0, 0}},
+				Skip:     true, // this fails merging right into left
```

Comment:
Interesting! 

I ran this locally and it looks like it's actually failing on the left-to-right merge with:
```
index key ( 101, 1 ) not found in index col1

Secondary index contents:
  - k: [1 0 0 0 0 0 2 0] 
  - k: [2 0 0 0 0 0 2 0] 
  - k: [3 0 0 0 0 0 2 0] 
  - k: [4 0 0 0 0 0 2 0] 
  - k: [5 0 0 0 0 0 2 0] 
  - k: [6 0 0 0 0 0 2 0] 
```

Looks like the index created after the merge only has `pk` and doesn't contain `col1` at all, right? 

A couple other clues I noticed while playing around with this one:
- when I changed both sides to have the same virtual column expression (`pk + 100`), it only failed when going from left-to-right, probably because left still specified the virtual column was `stored`. 
- when I changed both sides to be virtual, non-stored columns, with the same expression, it failed with: `panic: runtime error: index out of range [0] with length 0` (coming from `merge_prolly_rows.go` on line 93).

Reply:
Thanks, making a note of this